python/ray/experimental/workflow/workflow_access.py

```diff
@@ -0,0 +1,35 @@
+import logging
+from typing import Any
+
+import ray
+
+logger = logging.getLogger(__name__)
+
+
+class WorkflowExecutionError(Exception):
+    def __init__(self, workflow_id: str):
+        self.message = f"Workflow[id={workflow_id}] failed during execution."
+        super().__init__(self.message)
+
+
+@ray.remote
+def workflow_output_cache(workflow_id: str, output: ray.ObjectRef) -> Any:
```

Comment:
cache doesn't make sense as the verb here, how about `resolve_workflow_outputs` instead?